Database preparation scripts

rafraser opened the pull request:
This pull request contains two key SQL files and some example queries. For more information about the database structure, see the [Database Structure wiki page](https://github.com/summerprogram2019/ecosoft/wiki/Database-Structure)

- create.sql creates the required tables
- data.sql has some example testing data
- Sample queries for leaderboards and other useful functionality

Closes #6

rafraser committed:
Database creation script

rafraser committed:
Improved friends table comment

rafraser committed:
Created example data for DB testing

rafraser committed:
Example database queries

Example files that are used to test some useful database functionality. These include generating leaderboards, summing article likes, and checking for any erroneous one-way friendships.

Guthers commented on db/create.sql:
```diff
@@ -0,0 +1,88 @@
+
+CREATE TABLE IF NOT EXISTS `activities` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `name` tinytext NOT NULL,
```
Should use varchar and specify size

rafraser replied:
Resolved in 864b5d0

Guthers commented on db/create.sql:
```diff
@@ -0,0 +1,88 @@
+  `user1` int(11) NOT NULL,
+  `user2` int(11) NOT NULL,
+  PRIMARY KEY (`user1`,`user2`),
+  KEY `FK_friends_user_2` (`user2`),
```
Should there not also be a user_1 key?

rafraser replied:
Resolved in 1b50d70

Guthers commented on db/create.sql:
```diff
@@ -0,0 +1,88 @@
+
+CREATE TABLE IF NOT EXISTS `goals` (
+  `uid` int(11) NOT NULL,
+  `description` text NOT NULL,
```
again varchar?

rafraser replied:
Resolved in 864b5d0

Guthers commented on db/create.sql:
```diff
@@ -0,0 +1,88 @@
+
+CREATE TABLE IF NOT EXISTS `flowers` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `name` tinytext NOT NULL,
```
varchar?

rafraser replied:
Resolved in 864b5d0

Guthers commented on db/create.sql:
```diff
@@ -0,0 +1,88 @@
+CREATE TABLE IF NOT EXISTS `articles` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `title` text NOT NULL,
+  `description` text,
```
varchar?

rafraser replied:
Resolved in 864b5d0

Guthers commented on db/create.sql:
```diff
@@ -0,0 +1,88 @@
+
+CREATE TABLE IF NOT EXISTS `articles` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `title` text NOT NULL,
```
varchar?

rafraser replied:
Resolved in 864b5d0

Guthers commented on db/data.sql:
```diff
@@ -0,0 +1,92 @@
+INSERT INTO `user` (`uid`, `username`, `password`) VALUES
+	(1, 'rafraser', 'klagjaihsakg'),
```
Are these the plain text passwords? If not, what has being done to them?

rafraser replied:
Discussion in #35

Guthers commented on db/sample_queries/articlelikes.sql:
```diff
@@ -0,0 +1,4 @@
+SELECT a.title AS 'Article Name', COUNT(l.uid) AS 'Number Of Likes' 
```
Should add a commented out line at the start describing what the sample queries do

rafraser replied:
Resolved in 4fb0fe2

rafraser committed:
Changes to some DB types

- Tinytext changed to varchar
- int(11) is now int(12)
- T/F enums are now tinyint(1)
- Articles now store time of posting

rafraser committed:
Adjustments to foreign keys

rafraser committed:
Sample queries are commented
